tests/test_authorisation.py

```diff
@@ -351,9 +340,136 @@ def test_different_user_levels_have_correct_flagging_permissions(
                     "assess/flag/app123",
                     follow_redirects=True,
                 )
+                assert False, "Did not throw expected error."
```

Comment:
If you're interested, there is a better way to test expected exceptions using a built-in pytest context manager:

https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Here's an example I used this for before:

https://github.com/communitiesuk/funding-service-design-assessment/blob/02bc025a91a15e79c9d1d21951f08c298601f67a/tests/test_applicators_response.py#L373-L382